bionic/core/task_execution.py

```diff
@@ -751,13 +752,35 @@ def _local_artifact_from_value(self, value):
                 )
             ) from e
 
-        value_hash = protocol.tokenize_file(value_path)
+        value_hash = self._generate_value_hash(value, value_path, context)
         return Artifact(
             url=url_from_path(value_path),
             content_hash=value_hash,
         )
         return value_path
 
+    def _generate_value_hash(self, value, value_path, context):
```

Comment:
Should we do this for all values, or just pickled ones? For example, if we serialize a dataframe object to a Parquet file, it will always get deserialized as a standard Pandas dataframe, so it's not clear that the details of the original class matter.

Reply:
I like this idea!